internal/receiver/scriptedinputsreceiver/procpipe/procpipe.go

```diff
@@ -0,0 +1,273 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procpipe
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+const (
+	operatorType = "procpipe_input"
+
+	// minMaxLogSize is the minimal size which can be used for buffering
+	// TCP input
+	minMaxLogSize = 64 * 1024
+
+	// DefaultMaxLogSize is the max buffer sized used
+	// if MaxLogSize is not set
+	DefaultMaxLogSize = 1024 * 1024
+
+	DefaultIntervalSeconds = 60
+)
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+// NewConfig creates a new stdin input config with default values
+func NewConfig() *Config {
```

Comment:
I think all the config logic should be moved to `scriptedinputsreceiver/config.go` w/ added unit tests and the executor details in this package.